tasks/msapprovals/msapprovals.go

```diff
@@ -47,6 +46,13 @@ func (p *Task) ProcessMessages(ctx context.Context, ts *types.TipSet, pts *types
 		StateRoot: pts.ParentState().String(),
 	}
 
+	tsMsgs, err := p.node.GetExecutedAndBlockMessagesForTipset(ctx, ts, pts)
+	if err != nil {
+		report.ErrorsDetected = xerrors.Errorf("getting executed and block messages: %w", err)
+		return nil, report, nil
+	}
+	emsgs := tsMsgs.Executed
+
```

Comment:
(ditto) Returning the error here would be fatal meaning the corresponding walk, watch, or fill job would abort completely -- that would be inconsistent with current behavior. So, we list this as a detected error in the report which will mark the task as `ERROR` in the corresponding processing report, and return immediately with no data extracted.